### PR TITLE
docs(ts): fix typescript definitions

### LINKED ISSUE #2423: Old jsdoctypeparser version (via dox) can't handle typescript function types
### High-level description

ScalarBarActor.d.ts has a type that causes jsdoctypeparser v1.2.0 to fail:
```
          throw new Lexer.SyntaxError(e.message, typeExp, e.offset);
          ^
[Error [TypeLexerSyntaxError]: Expected "!", "#", ")", ".", ".<", "<", "=", "?", "[]", "|", "~", [ \t\r\n ] or [a-zA-Z0-9_$] but ":" found.:
(helper: any) => void
      ^]
```

### Steps to reproduce
- Run `npm run doc:generate-api`

### Detailed behavior

`dox` depends on `jsdoctypeparser@^1.2.0`, which fails at parsing `(helper: any) => void` in the param type of a jsdoc string [here](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/ScalarBarActor/index.d.ts#L197).

FYI @daker there is a PR to dox to upgrade jsdoctypeparser in dox (https://github.com/tj/dox/pull/196).

### Expected behavior

Generating the API does not error out.

### Environment
- **vtk.js**: 24.12.1
- **OS**: N/A
- **Browser**: N/A

### PR DESCRIPTION
**Pull request by daker**
fix #2423

### Changes

- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

- [ ] This change adds or fixes unit tests 
- [ ] Tested environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

**Comment by daker** [comment]
@Julesdevops or @floryst if you want to review it

File: Sources/Rendering/Core/Coordinate/Constants.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export declare enum Coordinate {
```
Comment (Julesdevops) [inline Sources/Rendering/Core/Coordinate/Constants.d.ts:1]: I would name it `CoordinateSystems` for consistency.

File: Sources/Rendering/Core/ImageMapper/Constants.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export declare enum SlicingMode {
```
Comment (Julesdevops) [inline Sources/Rendering/Core/ImageMapper/Constants.d.ts:1]: `SlicingModes` if we want to be consistent.

File: Sources/Rendering/Core/ImageProperty/Constants.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export declare enum InterpolationType {
```
Comment (Julesdevops) [inline Sources/Rendering/Core/ImageProperty/Constants.d.ts:1]: Same as above

**Review by Julesdevops — COMMENTED** [review]
Looks cool overall

File: Sources/Rendering/Core/ImageProperty/index.d.ts [hunk 4]
```diff
@@ -87,19 +83,19 @@ export interface vtkImageProperty extends vtkObject {
...
 	 */
-	getRGBTransferFunction(idx?: number): any;
+	getRGBTransferFunction(idx?: number): vtkColorTransferFunction;
 
 	/**
 	 * Alias to get the piecewise function (backwards compatibility)
 	 * @param {Number} [idx]
 	 */
-	getScalarOpacity(idx: number): number;
+	getScalarOpacity(idx?: number): any;
```
Comment (floryst) [inline Sources/Rendering/Core/ImageProperty/index.d.ts:92]: This should return a `vtkPiecewiseFunction`.

File: Sources/Rendering/Core/ImageMapper/Constants.d.ts (hunk 2 above)
Comment (daker) [reply Sources/Rendering/Core/ImageMapper/Constants.d.ts:1]: well it should SlicingModes but the Constants.js we have SlicingMode

File: Sources/Rendering/Core/Coordinate/Constants.d.ts (hunk 1 above)
Comment (daker) [reply Sources/Rendering/Core/Coordinate/Constants.d.ts:1]: That's what we have in Constants.js

Comment (Julesdevops) [reply Sources/Rendering/Core/Coordinate/Constants.d.ts:1]: fair enough

**Comment by floryst** [comment]
LGTM

**Pull request merged**